packages/frontend/src/pages/UserDetails/UserDetailsPage.tsx

```diff
@@ -0,0 +1,88 @@
+import { faUserGroup } from '@fortawesome/free-solid-svg-icons';
+import BreadCrumb from '@/components/BreadCrumb';
+import BackLink from '@/navigation/BackLink';
+import { SingleUser, SingleUserParams, useAdminUsers } from '@/model/users';
+import { useRecoilValue } from 'recoil';
+import { useParams } from 'react-router-dom';
+import { formatIsoDateUTC } from '@/utils/date';
+import { shortenEthAddress } from '@/utils/index';
+import { UserDto, UserRole } from 'api/dist/user/types';
+import { toast } from 'react-hot-toast';
+
+const roles = [UserRole.ADMIN, UserRole.FORWARDER, UserRole.QUANTIFIER];
+
+const UserDetailsPage = (): JSX.Element => {
+  const { userId } = useParams<SingleUserParams>();
+  const user = useRecoilValue(SingleUser({ userId }));
+  const { addRole, removeRole } = useAdminUsers();
+
+  const handleRole = (role: UserRole, user: UserDto): void => {
+    let resp;
+    const isRemove = user.roles.includes(role);
+    async (): Promise<void> => {
```

Comment:
It looks like you declare this function but it's never run? It doesn't seem to make any network requests when 'handleRole' is called.

Why not just modify `handleRole` to return a `Promise<void>`? Then you can get rid of this nested function.

Reply:
The ESlint pattern that we are currently using doesn't allow us to use a function that returns Promise<void> on the onClick properties, because they're expecting a void return. I can just disable [the rule](https://typescript-eslint.io/rules/no-misused-promises/) for the file, but I really don't know if this is a good idea.

Reply:
You can actually put a `void` in front of the function call and typescript will ignore the return type.

So this: `onClick={async (): Promise<void> => handleRole(role, user)}` would change to: `onClick={(): void => void handleRole(role, user)}`